plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/modules/code/DeadCodeHelper.java

```diff
@@ -565,7 +565,9 @@ public static void mergeBasicBlocks(ControlFlowGraph graph) {
           }
         }
       }
-
+      if(graph.getBlocks().size() == originBlocksCount && merged){
+        break;
+      }
       if (!merged) {
```

Comment:
`!merged || graph.getBlocks().size() == originBlocksCount` would convey the idea better IMO.

Reply:
I have changed the statement as you said.